src/hooks/useAuthStatus.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * 인증 상태 및 권한 관리 훅
+ */
+
+import { useAuthStore } from '@/stores/auth';
+import { mapStudyRoleToUserRole } from '@/utils';
+import type { UserRole, StudyRole } from '@/types';
+import {
+  hasPermission,
+  isAdmin,
+  isStudyLeader,
+  isStudyMember,
+  canAccess,
+} from '@/types';
+
+// 인증 상태 및 권한 관리 훅
+export const useAuthStatus = () => {
+  const { isLogin, isInitialized, user } = useAuthStore();
```

Comment:
```suggestion
  const { accessToken, isInitialized, user } = useAuthStore();
  const isLogin = !!accessToken;
```

isLogin은 별도의 상태로 관리하기보다, accessToken의 유무로 판단할 수 있는 파생값으로 처리하는건 어떨까요?
그럼 매번 isLogin 상태를 관리할 필요가 없어집니다.(관리해야 하는 상태가 줄어드는 장점) setAccessToken을 할 때, setIsLogin 처리를 깜빡하여 오류가 발생하는 등의 문제도 예방할 수 있고요.

때문에 accessToken은 SSOT로 삼고, isLogin은 파생값으로 처리하는걸 권장드립니다.

Reply:
점점 관리해야 하는 상태가 많아져서 이걸 어떻게 개선할 수 있을까 고민했었는데, 매우 좋은 개선 방법이라 생각합니다. 감사합니다!